### DISCUSSION TITLE
Query collection to compare against a numerical value

### DISCUSSION THREAD

**Discussion by hiralm21** [ravendb/ravendb#16502] [Q&A] [answered]
Hi!

I am trying to write RQL to query a collection to get all documents  - 
WHERE a <= b + 5

However, it throws an error -
Expected value, but got: <Field>: b

Am I missing anything?

**Comment by ayende** [comment]
Yes, the problem is that this force us to evaluate this in runtime.
Instead, use an index that work on `DiffOfA_B = A - B` and do a query on that.

**Reply by hiralm21** [reply]
@ayende - So this is what I want the whole query to look like 
- from collection
- where type = 'some specific type'
- and case when a <= 10 then a between (b * 0.1) - 1 and  (b * 0.1)  + 1 
- else a between b -1 and b + 1
- and c between d -20 and d + 20

and so on... (please ignore the syntax)
so there are multiple conditions involved. I can create index on a-b but how do I incorporate that into my query> Also, I have multiple indexes to create on c and d and so on?  Apologies, I am new to RavenDB hence all the questions

**Reply by ayende** [reply, marked as answer]
Put those logics in the index, as separate clauses, something like:

```
from doc in docs.MyItems
select new
{
   ImportantItem = doc.A <= 10 && doc.A >= (doc.B *01 -1) && doc.A < (doc.B * 0.1+1),
   NiceItem = doc.A >= doc.B+1 && doc.A < (doc.B -1)
}
```

When you query: `from index MyIndex where ImportItem = true`

That means that the computation happens at indexing time and queries are *really* fast.

**Reply by hiralm21** [reply]
Thank you so much @ayende ! It worked for our use case. Follow up question - can we query from 2 indexes together? like load an index while querying another one?

**Reply by ayende** [reply]
You can load a *document*, not an index, during a query

I recommend creating a new question with more details, I don't think that I understand you here

**Reply by hiralm21** [reply]
Will do. Thank you for your help! :)
